core/src/tangram.cpp

```diff
@@ -887,6 +887,18 @@ void Map::runAsyncTask(std::function<void()> _task) {
     }
 }
 
+void Map::onMemoryWarning() {
+    auto& tileCache = impl->tileManager.getTileCache();
+
+    if (tileCache) {
+        tileCache->clear();
+    }
+
+    if (impl->scene && impl->scene->fontContext()) {
+        impl->scene->fontContext()->releaseFonts();
```

Comment:
It will reload/load the font whenever a URI or a font handle loader has been given to the font context, this happens whenever it is needed again from the font context to render a glyph (not if the data was given by a `char*` where the fonts couldn't be loaded again), so no scene init involved in that process. 

You can trigger a memory warning using F3 in this branch to see how it goes through the process.